internal/services/containers/container_registry_task_resource.go

```diff
@@ -0,0 +1,1551 @@
+			Name:     utils.String(k),
+			Value:    utils.String(v),
+			IsSecret: utils.Bool(true),
+		})
+	}
+	return &out
+}
+
+func flattenRegistryTaskValues(values *[]legacyacr.SetValue) map[string]string {
+	if values == nil {
+		return nil
+	}
+
+	vals := map[string]string{}
+
+	for _, value := range *values {
+		var (
+			k        string
+			v        string
+			isSecret bool
+		)
+
+		if value.Name != nil {
+			k = *value.Name
+		}
+		if value.Value != nil {
+			v = *value.Value
+		}
+		if value.IsSecret != nil {
+			isSecret = *value.IsSecret
+		}
+
+		if isSecret {
+			// GET response won't return the value of secret values
+			continue
+		}
+		vals[k] = v
+	}
+	return vals
+}
+
+func expandRegistryTaskIdentity(identities []Identity) *legacyacr.IdentityProperties {
+	if len(identities) == 0 {
+		return nil
+	}
+
+	identity := identities[0]
+	out := &legacyacr.IdentityProperties{
+		Type: legacyacr.ResourceIdentityType(identity.Type),
+	}
+	if len(identity.IdentityIds) > 0 {
+		out.UserAssignedIdentities = map[string]*legacyacr.UserIdentityProperties{}
+		for _, identityId := range identity.IdentityIds {
+			out.UserAssignedIdentities[identityId] = &legacyacr.UserIdentityProperties{}
+		}
+	}
+	return out
+}
+
+func flattenRegistryTaskIdentity(identity *legacyacr.IdentityProperties) ([]Identity, error) {
+	if identity == nil {
+		return nil, nil
+	}
+
+	obj := Identity{
+		Type: string(identity.Type),
+	}
+
+	if identity.PrincipalID != nil {
+		obj.PrincipalId = *identity.PrincipalID
+	}
+	if identity.TenantID != nil {
+		obj.TenantId = *identity.TenantID
+	}
+
+	var identityIds []string
+	for id := range identity.UserAssignedIdentities {
+		identityId, err := msiparse.UserAssignedIdentityIDInsensitively(id)
+		if err != nil {
+			return nil, fmt.Errorf("parsing identity id %s: %w", id, err)
+		}
+		identityIds = append(identityIds, identityId.ID())
+	}
+	obj.IdentityIds = identityIds
+
+	return []Identity{obj}, nil
+}
+
+func expandRegistryTaskPlatform(input []Platform) *legacyacr.PlatformProperties {
+	if len(input) == 0 {
+		return nil
+	}
+	platform := input[0]
+	out := &legacyacr.PlatformProperties{
+		Os: legacyacr.OS(platform.OS),
+	}
+	if arch := platform.Architecture; arch != "" {
+		out.Architecture = legacyacr.Architecture(arch)
+	}
+	if variant := platform.Variant; variant != "" {
+		out.Variant = legacyacr.Variant(variant)
+	}
+	return out
+}
+
+func flattenRegistryTaskPlatform(platform *legacyacr.PlatformProperties) []Platform {
+	if platform == nil {
+		return nil
+	}
+	return []Platform{{
+		OS:           string(platform.Os),
+		Architecture: string(platform.Architecture),
+		Variant:      string(platform.Variant),
+	}}
+}
+
+func expandRegistryTaskCredentials(input []RegistryCredential) *legacyacr.Credentials {
+	if len(input) == 0 {
+		return nil
+	}
+
+	out := legacyacr.Credentials{
+		SourceRegistry:   &legacyacr.SourceRegistryCredentials{LoginMode: legacyacr.SourceRegistryLoginModeDefault},
```

Comment:
I see in the API docs that there's also another login mode called "None" (I'm assuming it means it should not try to autenticate at all?). 

Are we skipping it on purpose?